paasta_tools/kubernetes_tools.py

```diff
@@ -3502,22 +3502,26 @@ def get_kubernetes_secret(
     secret_name: str,
     service_name: str,
     namespace: str = "paasta",
-) -> str:
+    decode: bool = True,
+) -> Union[str, bytes]:
 
     k8s_secret_name = get_kubernetes_secret_name(service_name, secret_name)
 
     secret_data = kube_client.core.read_namespaced_secret(
         name=k8s_secret_name, namespace=namespace
     ).data[secret_name]
-    secret = base64.b64decode(secret_data).decode("utf-8")
-    return secret
+    # String secrets (e.g. yaml config files) need to be decoded
+    # Binary secrets (e.g. TLS Keystore or binary certificate files) cannot be decoded
+    if decode:
+        return base64.b64decode(secret_data).decode("utf-8")
+    return base64.b64decode(secret_data)
 
 
 def get_kubernetes_secret_env_variables(
     kube_client: KubeClient,
     environment: Dict[str, str],
     service_name: str,
-) -> Dict[str, str]:
+) -> Dict[str, Union[str, bytes]]:
```

Comment:
hmm, we're in a bit of a pickle here - i'm pretty sure env vars must be strings

i think we're always going to get strings here (`get_kubernetes_secret()` will decode by default) so we're actually fine, but I don't think there's really a way to get mypy to follow the logic and know that `get_kubernetes_secret()` will always return a string if `decode` is true

not really sure what the best path forward is here, but my current thought is:
* `Dict[str, Union[str, bytes]]` -> Dict[str, str]
* explicitly call `get_kubernetes_secret()` with `decode=True` + maybe a comment 
* cast the call to `get_kubernetes_secret()` to `str` (to get mypy to be happy with the return value typing)

Reply:
Obliging these suggestions as stated